core/policy/repository.go

```diff
@@ -182,9 +182,9 @@ func (r *Repository) IsHostAllowed(host string) bool {
 }
 
 func (r *Repository) findItemFor(policy market.AccessPolicy) (*listItem, error) {
-	for _, item := range r.items {
+	for i, item := range r.items {
 		if item.policy == policy {
-			return &item, nil
+			return &r.items[i], nil
```

Comment:
Isn't `r.items[i]` is the same as `item` ? How did this helped?

Reply:
Its not, range produces a copy.